cmd/blockchaincmd/deploy.go

```diff
@@ -534,14 +535,14 @@ func deployBlockchain(cmd *cobra.Command, args []string) error {
 					}
 				}
 				network = models.NewNetworkFromCluster(network, clusterName)
-				nodeConfig := ""
+				nodeConfig := map[string]interface{}{}
 				if app.AvagoNodeConfigExists(blockchainName) {
-					nodeConfigBytes, err := os.ReadFile(app.GetAvagoNodeConfigPath(blockchainName))
+					nodeConfig, err = utils.ReadJSON(app.GetAvagoNodeConfigPath(blockchainName))
 					if err != nil {
 						return err
 					}
-					nodeConfig = string(nodeConfigBytes)
 				}
+				nodeConfig[config.PartialSyncPrimaryNetworkKey] = true
```

Comment:
Taking a step back, is there a more general place for us to set the partial sync? 

We should set it for any node that will be an L1 validator only, no matter if that node is run locally or remotely. And conversely, if we want to set up a local node to be a primary network validator, then we won't want this flag set.

Reply:
I can add flags to `blockchain deploy` and `node local start` to specify if primary network validation is desired
(partial sync being default). that flags to be applied both to cluster creation, and to cluster restart. That following
current scheme.
I did not considered too much on generalities as remote node setup is currently under discussion, and fuji
primary setup is going to take other bootstrap time and was not a target.

Reply:
but if generality is desired so, we can add either a boolean field for partial sync to the cluster definition struct (if we want to focus on that setting being a relevant part of our configuration), or a field for flags in general to the cluster definition struct (if we want to enable general flexible changes to a cluster definition). I tend to prefer later one as it enables changes in the future with less code changes (we just set/change flags).
in that case, previous to starting the local cluster for the first time, the cluster def is created in a previous, 
separate step. so probably the main function called from inside `blockchain deploy` and `node local start` will first call a func to create the cluster def and then a second func to execute the cluster. the second function, probably being also the function that restarts a cluster, will check if the cluster is partial sync or not, and set the correct flag.
if the use later on wants this cluster to be full sync, for example or restart, a flag should be set will should be used to change the cluster conf probably previous restart.

Reply:
partial sync for a remote node should probably be target of another PR

Reply:
Sorry I'm not completely following your suggestions, but I think we should always strive to ensure things are as general/modular as possible. 🙂 

In this case, whether or not to set the `--partial-sync-primary-network` flag should be a determination that is made whenever we're starting/configuring an AvalancheGo node to be run, no matter where it's going to be executed.

I'm not exactly sure how that fits into the commands it should apply to, etc, but having it only apply to local nodes I think is likely to result in either duplicating it in the future in other place or needing to rework it to ensure it applies to all of the correct cases.

Reply:
added cmdline flags for setting partial sync both for local and remote. as can be verified on https://github.com/ava-labs/avalanche-cli/pull/2309/commits/44b6dab72817a2eeddd92965f86c7069fa2ea057 the flows and ways to configure nodes are pretty different among local and remote. also remote has some issues related to node configuration (two different pieces need to be modified when changing something, one for start and another for restart). refactoring needed to fix and unify this most probably excess the scope of this PR.

Reply:
cmdline flags are set for partial sync turned on by default